### PR TITLE
Remove telemetry fields from Experiment Model 

### LINKED ISSUE #3008: Remove telemetry fields from Experiment model
I think the telemetry field definitions will be statically defined in the nimbus shared repo:

https://github.com/mozilla/nimbus-shared/blob/main/data/features/picture_in_picture.toml#L10-L12

so there's no need to store them on the experiment model.

### PR DESCRIPTION
**Pull request by bhushan-borole**
Fixes #3008

### CODE REVIEW

**Comment by bhushan-borole** [comment]
Hello @jaredlockhart 
Can you please help me out in this error
https://app.circleci.com/pipelines/github/mozilla/experimenter/5425/workflows/ab50c279-a8aa-497e-9361-175b135e1193/jobs/14215
Thank you

**Comment by bhushan-borole** [comment]
Aah I knew it, I missed on those references.
Thank you @jaredlockhart will check on that.

**Comment by bhushan-borole** [comment]
Hello @jaredlockhart 
Do I have to remove the fields from here also ?
https://github.com/mozilla/experimenter/blob/6b46cbea856e96e6596baf0818ef52ac2f86fb11/app/experimenter/templates/experiments/section_analysis.html#L23

**Comment by jaredlockhart** [comment]
Exactly, everywhere you find a reference to them in models, forms, serializers, templates

**Comment by bhushan-borole** [comment]
Done @jaredlockhart !, removed all the references and tests ran successfully.

File: app/experimenter/experiments/forms.py [hunk 1]
```diff
@@ -294,21 +294,6 @@ class ExperimentObjectivesForm(ChangeLogMixin, forms.ModelForm):
...
-        required=False,
-        label="Telemetry Conversion Event Method",
-        help_text=Experiment.TELEMETRY_EVENT_HELP_TEXT,
-    )
-    telemetry_event_object = forms.CharField(
-        required=False,
-        label="Telemetry Conversion Event Object",
-        help_text=Experiment.TELEMETRY_EVENT_HELP_TEXT,
-    )
     telemetry_event_value = forms.CharField(
```
Comment (jaredlockhart) [inline app/experimenter/experiments/forms.py:297]: Don't forget this one too!

File: app/experimenter/experiments/forms.py [hunk 2]
```diff
@@ -341,9 +326,6 @@ class Meta:
             "objectives",
             "analysis",
             "total_enrolled_clients",
-            "telemetry_event_category",
-            "telemetry_event_object",
-            "telemetry_event_method",
             "telemetry_event_value",
```
Comment (jaredlockhart) [inline app/experimenter/experiments/forms.py:329]: And here

File: app/experimenter/experiments/migrations/0096_add_telemetry.py [hunk 3]
```diff
@@ -10,21 +10,6 @@ class Migration(migrations.Migration):
...
-        migrations.AddField(
-            model_name="experiment",
-            name="telemetry_event_method",
-            field=models.CharField(blank=True, max_length=255, null=True),
-        ),
-        migrations.AddField(
-            model_name="experiment",
-            name="telemetry_event_object",
-            field=models.CharField(blank=True, max_length=255, null=True),
-        ),
```
Comment (jaredlockhart) [inline app/experimenter/experiments/migrations/0096_add_telemetry.py:27]: Oh sorry I should've mentioned, the only place you don't remove it is in this file

File: app/experimenter/experiments/api/v1/serializers.py [hunk 4]
```diff
@@ -138,9 +138,6 @@ class Meta:
             "variants",
             "results",
             "changes",
-            "telemetry_event_category",
-            "telemetry_event_method",
-            "telemetry_event_object",
             "telemetry_event_value",
```
Comment (jaredlockhart) [inline app/experimenter/experiments/api/v1/serializers.py:141]: Here too

File: app/experimenter/experiments/changelog_utils.py [hunk 5]
```diff
@@ -70,9 +70,6 @@ class Meta:
             "total_enrolled_clients",
             "analysis",
             "analysis_owner",
-            "telemetry_event_category",
-            "telemetry_event_method",
-            "telemetry_event_object",
             "telemetry_event_value",
```
Comment (jaredlockhart) [inline app/experimenter/experiments/changelog_utils.py:73]: And here

File: app/experimenter/experiments/models.py [hunk 6]
```diff
@@ -236,9 +236,6 @@ class Experiment(ExperimentConstants, models.Model):
 
     data_science_issue_url = models.URLField(blank=True, null=True)
     feature_bugzilla_url = models.URLField(blank=True, null=True)
-    telemetry_event_category = models.CharField(max_length=255, blank=True, null=True)
-    telemetry_event_method = models.CharField(max_length=255, blank=True, null=True)
-    telemetry_event_object = models.CharField(max_length=255, blank=True, null=True)
     telemetry_event_value = models.CharField(max_length=255, blank=True, null=True)
```
Comment (jaredlockhart) [inline app/experimenter/experiments/models.py:239]: And here

File: app/experimenter/experiments/tests/api/v1/test_serializers.py [hunk 7]
```diff
@@ -151,9 +151,6 @@ def test_serializer_outputs_expected_schema(self):
                 "results_measure_impact": None,
                 "results_impact_notes": None,
             },
-            "telemetry_event_category": experiment.telemetry_event_category,
-            "telemetry_event_method": experiment.telemetry_event_method,
-            "telemetry_event_object": experiment.telemetry_event_object,
             "telemetry_event_value": experiment.telemetry_event_value,
```
Comment (jaredlockhart) [inline app/experimenter/experiments/tests/api/v1/test_serializers.py:154]: And here

File: app/experimenter/experiments/tests/factories.py [hunk 8]
```diff
@@ -80,9 +80,6 @@ class ExperimentFactory(ExperimentConstants, factory.django.DjangoModelFactory):
     )
     objectives = factory.LazyAttribute(lambda o: faker.text(1000))
     analysis = factory.LazyAttribute(lambda o: faker.text(1000))
-    telemetry_event_category = factory.LazyAttribute(lambda o: faker.word())
-    telemetry_event_method = factory.LazyAttribute(lambda o: faker.word())
-    telemetry_event_object = factory.LazyAttribute(lambda o: faker.word())
     telemetry_event_value = factory.LazyAttribute(lambda o: faker.word())
```
Comment (jaredlockhart) [inline app/experimenter/experiments/tests/factories.py:83]: Here

File: app/experimenter/experiments/tests/test_changelog_utils.py [hunk 9]
```diff
@@ -65,9 +65,6 @@ def test_serializer_outputs_expected_schema(self):
             "total_enrolled_clients": experiment.total_enrolled_clients,
             "analysis": experiment.analysis,
             "analysis_owner": experiment.analysis_owner.id,
-            "telemetry_event_category": experiment.telemetry_event_category,
-            "telemetry_event_method": experiment.telemetry_event_method,
-            "telemetry_event_object": experiment.telemetry_event_object,
             "telemetry_event_value": experiment.telemetry_event_value,
```
Comment (jaredlockhart) [inline app/experimenter/experiments/tests/test_changelog_utils.py:68]: Here

**Comment by jaredlockhart** [comment]
Sorry I should've left clearer instructions on the ticket, what we needed removed are all four of:

            telemetry_event_category
            telemetry_event_method
            telemetry_event_object
            telemetry_event_value

File: app/experimenter/experiments/forms.py (hunk 1 above)
Comment (bhushan-borole) [reply app/experimenter/experiments/forms.py:297]: Fixed

File: app/experimenter/experiments/forms.py (hunk 2 above)
Comment (bhushan-borole) [reply app/experimenter/experiments/forms.py:329]: Fixed

File: app/experimenter/experiments/migrations/0096_add_telemetry.py (hunk 3 above)
Comment (bhushan-borole) [reply app/experimenter/experiments/migrations/0096_add_telemetry.py:27]: Fixed

File: app/experimenter/experiments/api/v1/serializers.py (hunk 4 above)
Comment (bhushan-borole) [reply app/experimenter/experiments/api/v1/serializers.py:141]: Fixed

File: app/experimenter/experiments/changelog_utils.py (hunk 5 above)
Comment (bhushan-borole) [reply app/experimenter/experiments/changelog_utils.py:73]: Fixed

File: app/experimenter/experiments/models.py (hunk 6 above)
Comment (bhushan-borole) [reply app/experimenter/experiments/models.py:239]: Fixed

File: app/experimenter/experiments/tests/api/v1/test_serializers.py (hunk 7 above)
Comment (bhushan-borole) [reply app/experimenter/experiments/tests/api/v1/test_serializers.py:154]: Fixed

Comment (bhushan-borole) [reply app/experimenter/experiments/tests/api/v1/test_serializers.py:154]: Fixed

File: app/experimenter/experiments/tests/factories.py (hunk 8 above)
Comment (bhushan-borole) [reply app/experimenter/experiments/tests/factories.py:83]: Fixed

File: app/experimenter/experiments/tests/test_changelog_utils.py (hunk 9 above)
Comment (bhushan-borole) [reply app/experimenter/experiments/tests/test_changelog_utils.py:68]: Fixed

**Comment by bhushan-borole** [comment]
> Sorry I should've left clearer instructions on the ticket, what we needed removed are all four of:
> 
> ```
>         telemetry_event_category
>         telemetry_event_method
>         telemetry_event_object
>         telemetry_event_value
> ```

No problem, I removed all 4 fields.
Thank you for clarification.

**Comment by bhushan-borole** [comment]
I am facing a small issue here.
While running `make makemigrations`, my VM is crashing. I don't know what the issue is.

**Comment by jaredlockhart** [comment]
Oh! You generated the migration correctly, all you have left to do is run `make code_format` to reformat the code to pass linting and you're done!  This looks great!

**Comment by bhushan-borole** [comment]
> Oh! You generated the migration correctly, all you have left to do is run `make code_format` to reformat the code to pass linting and you're done! This looks great! You can run the full test suite locally with `make check`. I just tried it after running `make code_format` and everything passed perfectly so just push up the format changes and that's it :D

Thank you @jaredlockhart 
All the tests ran successfully.

File: app/experimenter/experiments/tests/api/v1/test_serializers.py [hunk 10]
```diff
@@ -150,11 +150,8 @@ def test_serializer_outputs_expected_schema(self):
                 "results_confidence": None,
                 "results_measure_impact": None,
                 "results_impact_notes": None,
+                "results_impact_notes": None,
```
Comment (jaredlockhart) [inline app/experimenter/experiments/tests/api/v1/test_serializers.py:153]: Oops duplicate line!

Comment (bhushan-borole) [reply app/experimenter/experiments/tests/api/v1/test_serializers.py:153]: Oh okay sorry didn't see that, will fix it

**Comment by bhushan-borole** [comment]
> Oh! I didn't realize, you're committing into your main branch in your fork, that's not exactly right and now it won't let me check out your pr locally.
> 
> In the [contributing instructions](https://github.com/mozilla/experimenter/blob/main/contributing.md#pull-request-process), step #3 is to create a branch in your local fork, and that's where the pull request comes from. We should be able to skip it for this PR but in the future, please create a branch on your local fork.
> 
> This looks great, just remove that duplicate line and this should be ready to land. Thanks again @bhushan-borole !! 😸

From next time surely, I'll keep it in mind.

**Comment by pdehaan** [comment]
Awesome stuff!! Thanks @bhushan-borole! 🎉

**Comment by bhushan-borole** [comment]
> Awesome stuff!! Thanks @bhushan-borole! 🎉 
> 

Thank you!!
Looking forward to contribute more!

**Pull request merged**